vms/platformvm/state/subnet_id_node_id_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/thepudds/fzgen/fuzzer"
+)
+
+func FuzzSubnetIDNodeIDMarshal(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		require := require.New(t)
+
+		var v subnetIDNodeID
+		fz := fuzzer.NewFuzzer(data)
+		fz.Fill(&v)
+
+		marshalledData := v.Marshal()
+
+		var parsed subnetIDNodeID
+		require.NoError(parsed.Unmarshal(marshalledData))
+		require.Equal(v, parsed)
+	})
+}
+
+func FuzzSubnetIDNodeIDUnmarshal(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		require := require.New(t)
+
+		var v subnetIDNodeID
+		if err := v.Unmarshal(data); err != nil {
+			require.ErrorIs(err, errUnexpectedSubnetIDNodeIDLength)
+			return
+		}
+
+		marshalledData := v.Marshal()
+		require.Equal(data, marshalledData)
+	})
+}
+
+func FuzzSubnetIDNodeIDMarshalOrdering(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var (
+			v0 subnetIDNodeID
+			v1 subnetIDNodeID
+		)
+		fz := fuzzer.NewFuzzer(data)
+		fz.Fill(&v0, &v1)
+
+		if v0.subnetID == v1.subnetID {
```

Comment:
is this return statement and if condition essential?

Reply:
If the subnetIDs are equal, then the comparison depends on the nodeIDs as well.